File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map <int, int> map;
        vector<int> ans;
        for (int i = 0; i < nums.size(); i++) {
            map[nums[i]] = i;
        }
        for (int i = 0; i < nums.size(); i++) {
            if (map.count(target - nums[i]) && map[target - nums[i]] != i) {
                ans.push_back(i);
                ans.push_back(map[target - nums[i]]);
                break;
            }
        }

        return ans;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        count = defaultdict(int)
        l = 0
        ans = 0

        for r, x in enumerate(s):
            count[x] += 1
            while (count[x] == 2):
                count[s[l]] -= 1
                l += 1
            ans = max(ans, r - l + 1)

        return ans